src/coreclr/jit/flowgraph.cpp

```diff
@@ -4120,6 +4120,80 @@ FlowGraphDfsTree* Compiler::fgComputeDfs()
 template FlowGraphDfsTree* Compiler::fgComputeDfs<false>();
 template FlowGraphDfsTree* Compiler::fgComputeDfs<true>();
 
+//------------------------------------------------------------------------
+// fgComputeLoopAwareDfs: Compute a depth-first search tree for the flow graph
+// where in the RPO traversal, loop bodies are visited before loop successors.
+//
+// Returns:
+//   The tree.
+//
+// Notes:
+//   If the flow graph has loops, the DFS will be reordered such that loop bodies are compact.
+//   This will invalidate BasicBlock::bbPreorderNum and BasicBlock::bbPostorderNum.
```

Comment:
I don't think we have any dependencies on `bbPreorderNum` or `bbPostorderNum` in the backend, but if we want to use loop-aware RPOs elsewhere in the JIT, I can work on making these members consistent.